Source/WPEFramework/PluginServer.h

```diff
@@ -3609,37 +3609,61 @@ POP_WARNING()
                     Core::ProxyType<Web::Response> response;
 
                     if (_jsonrpc == true) {
-                        if(_request->Verb == Request::HTTP_POST){
-                            Core::ProxyType<Core::JSONRPC::Message> message(_request->Body<Core::JSONRPC::Message>());
-                            if (HasService() == true) {
-                                message->ImplicitCallsign(GetService().Callsign());
+                        if(_request->Verb == Request::HTTP_POST) {
+                            Core::ProxyType<Web::JSONRPC::Body> message(_request->Body<Web::JSONRPC::Body>());
+                            if (message->Validity().IsSet() == true) {
+                                // Looks like we have a corrupted message.. Respond if posisble, with an error
+                                response = IFactories::Instance().Response();
+
+                                // If we also do not have an id, we can not return a suitable JSON message!
+                                if (message->Recorded().IsSet() == false) {
+                                    response->ErrorCode = Web::STATUS_BAD_REQUEST;
+                                    response->Message = _T("JSON-RPC was incorrectly formatted, could not deduce the id");
+                                }
+                                else {
+                                    uint32_t id = message->Recorded().Value();
+                                    string error = message->Validity().Value().Message();
+                                    response->ErrorCode = Web::STATUS_NO_CONTENT;
```

Comment:
204 is not a suitable response code - as per HTTP spec a 204 response cannot contain a body

>  A 204 response is terminated by the end of the header section; it cannot contain content or trailers.

Reply:
Suggest any other code. I did not read the spec, and happy with any value, just think that 200 is definitely not correct as something is not good (it never reached the JSONRPC logic)

Reply:
It automatically became Web::STATUS_BAD_REQUEST, since both with or without ID (passed back "id": null) it also return for the RESTFul call: Web::STATUS_BAD_REQUEST. Again did not read spec, so we can make it anything you like :-)